HotelBooking/Views/checked_in_rooms_view.py

```diff
@@ -94,58 +149,11 @@ def check_out_customer(self):
             if room_id != "Back":
                 self.room_controller.check_out_room(room_id)
                 print("\nSuccess!\n")
-            self.show()
-
-    def new_check_in(self):
```

Comment:
By removing this block, you're removing the ability to check in a walk-in customer who does not have a previous reservation. This is actually against the use-case diagram. Please add it back.

Reply:
Why not keep the check-in method exposed to the admin view and do the reserve & check-in in the exposed method behind the scene? In my opinion, it's better than asking the admin to perform 2 operations.

Reply:
If we are using the old way, it will not leave a record of the stay. (I need to track all the reservation history)

Reply:
I'm not saying we do the old way. I'm saying keep a method on the appropriate controller called `check_in_walk_in_customer` for `check_in_view`, and `check_in_walk_in_customer` will do:

1. make an immediate reservation 
2. modify this reservation to check-in status

Reply:
But It is also possable that hotel staff only want to make a reservation for a customer. I think it is better to leave it in two different functions

Reply:
I know that, but the hotel staff should be able to just "check-in" a walk-in customer directly, the controller will do the reservation modification behind the scenes. I am not saying to delete the reservation method, I'm saying to create the option of check-in flow for `check_in_walk_in_customer`.

Reply:
The old version, an admin could reserve a room for a customer AND check-in a customer without a reservation directly. In your current version, an admin can reserve a room for a customer, or check-in by first reserving and then modifying that reservation. I'm saying to create a new check-in for walk-in customers flow that asks the customer for the same parameters, but the method `check_in_walk_in_customer` would do the reservation+modify reservation.

Reply:
If so, it is kind of redundant that a hotel stuff have the option to check-in customer, make reservation for customer, and there is also an option to check-in and make reservation.

Reply:
Your check-in customer is checking-in customers with a reservation. For an admin to check-in a walk-in customer, they have to navigate to reservation, then navigate back, and then checking that reservation in. 
Create an operation dedicated to serve walk-in customers and save that back and forth operation.

Reply:
Actually, you don't have to add it if you don't want to. Sorry for being picky on this, I keep forgetting this is for school and not for work.

Reply:
We don't have real users of the app anyway, it is fine to be unclear in the UI I guess.

Reply:
just that I need to add a complete new view, which means that I also need to modify my diagram...

Reply:
NVM the request then (even though I don't see why you need to add a new view instead of just a new select option in check_in_rooms_view).

Reply:
Currently I am using the "reserved_rooms" view for both the customer and admin side, but since check-in is the functionality that only admin can have, I need to seprate the "reserved_rooms" for the customer and the admin.

Reply:
If it causes more breakage then forget it. Case closed.